projects/cm_mcu/FireFlyTask.c

```diff
@@ -702,6 +768,186 @@ void FireFlyTask(void *parameters)
       tmp2.us = data[0]; // change from uint_8 to int8_t, preserving bit pattern
       ff_status[ff].status = tmp2.s;
 
+      // Read the serial number
+#ifdef DEBUG
+      data[0] = 0x0U;
+      data[1] = 0x0U;
+      for (uint8_t i = 189; i < 205; i++) {// change from 171-185 to 189-198 or 189-204 or 196-211
+    	  r = SMBusMasterI2CWriteRead(smbus, ff_i2c_addrs[ff].dev_addr, &i, 1, data, 1);
+
+    	  if (r != SMBUS_OK) {
+    		  snprintf(tmp, 64, "FIF: %s: SMBUS failed (master/bus busy, ps=%d,c=%d)\r\n", __func__, ff,
+    				  2);
+    		  DPRINT(tmp);
+    		  continue; // abort reading this register
+    	  }
+    	  while (SMBusStatusGet(smbus) == SMBUS_TRANSFER_IN_PROGRESS) {
+    		  vTaskDelayUntil(&ff_updateTick, pdMS_TO_TICKS(10)); // wait
+    	  }
+    	  if (*p_status != SMBUS_OK) {
+    		  snprintf(tmp, 64, "FIF: %s: Error %d, break loop (ps=%d,c=%d) ...\r\n", __func__,
+    				  *p_status, ff, 2);
+    		  DPRINT(tmp);
+    		  ff_status[ff].serial_num[i - 196] = 3;
+    		  break;
+    	  }
+    	  convert_8_t tmp5;
+    	  tmp5.us = data[0]; // change from uint_8 to int8_t, preserving bit pattern
+    	  ff_status[ff].serial_num[i - 196] = tmp5.s;
+      }
+#endif
+
+      // Check the loss of signal alarm
+      data[0] = 0x0U;
```

Comment:
There is a lot of copy-paste here. Please change this into a  loop where all you change is which address is read.